src/main/java/com/stripe/net/APIRequest.java

```diff
@@ -0,0 +1,33 @@
+package com.stripe.net;
+
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+public class ApiRequest {
+  private BaseAddress baseAddress;
+  private ApiResource.RequestMethod method;
+  private String path;
+  private Map<String, Object> params;
```

Comment:
Params as a map won't work for RawRequest where params is a string.

Reply:
Do you think we should make the caller convert to a string or just provide multiple constructors?

Reply:
I'm not sure but having callers converting params to string will go agains our goal of centralizing the logic in ApiRequestor.

Reply:
Here is what I want to do re: RawRequest: https://github.com/stripe/stripe-java/pull/1717/commits/80833c9cc9e1c05dfbd5016ab03c3513f828fdf7#diff-2b360bd267e306d7c356360cb81b1acf0198ab73539a4a46ea5308824866bc57

I don't think there is much advantage as it stands to trying to represent the arguments to `ResponseGetter.rawRequest` and `ResponseGetter.request` with a single type -- separate types will do.

I was allured for a bit by the idea of letting users who want to implement StripeResponseGetter to provide only one method and not have to implement both `.rawRequest` and `.request`, but I don't think this is actually workable.